File: my-folder/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix=strs[0]
        for i in range(1,len(strs)):
            while strs[i].find(prefix)!=0:
                length=len(prefix)
                prefix=prefix[0:length-1]
                if len(prefix)==0:
                    return ""
        return prefix   

File: my-folder/0014-longest-common-prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n = strs.length;
        if(n == 0) return "";
        String prefix = strs[0];
        for(int i = 1; i < n; i++) {
            while(strs[i].indexOf(prefix) != 0) {
                prefix = prefix.substring(0, prefix.length()-1);
                if(prefix.isEmpty()) return "";
            }
        }
        return prefix;
    }
}
